flink-kubernetes-webhook/src/main/java/org/apache/flink/kubernetes/operator/admission/FlinkValidator.java

```diff
@@ -20,41 +20,74 @@
 import org.apache.flink.kubernetes.operator.admission.admissioncontroller.NotAllowedException;
 import org.apache.flink.kubernetes.operator.admission.admissioncontroller.Operation;
 import org.apache.flink.kubernetes.operator.admission.admissioncontroller.validation.Validator;
+import org.apache.flink.kubernetes.operator.crd.CrdConstants;
 import org.apache.flink.kubernetes.operator.crd.FlinkDeployment;
+import org.apache.flink.kubernetes.operator.crd.FlinkSessionJob;
 import org.apache.flink.kubernetes.operator.validation.FlinkResourceValidator;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
-import io.fabric8.kubernetes.api.model.GenericKubernetesResource;
+import io.fabric8.kubernetes.api.model.GroupVersionKind;
+import io.fabric8.kubernetes.api.model.KubernetesResource;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.Optional;
 import java.util.Set;
 
 /** Validator for FlinkDeployment creation and updates. */
-public class FlinkValidator implements Validator<GenericKubernetesResource> {
+public class FlinkValidator implements Validator<KubernetesResource> {
```

Comment:
Why do we need the change GenericKubernetesResource -> KubernetesResource???

Reply:
When I test, I found that the Object passed here may not always be `GenericKubernetesResource`. It may be the `CustomResource` or `GenericKubernetesResource`. The object retrieved from the AdmissionRequest is the `KubernetesResource` . So I think we do not have to assume that is `GenericKubernetesResource`. Please correct me if I'm wrong

Reply:
Bit confused, it's either FlinkDeployment of FlinkSessionjob that we're validating here, so I would expect a CustomResource here @gyfora?

Reply:
But I'm not sure what's the actual type of the `AdmissionRequest#object` . It's only defined as `KubernetesResource`